packages/fastify-react/plugin/index.js

```diff
@@ -12,26 +12,18 @@ import { closeBundle } from './preload.js'
 import { parseStateKeys } from './parsers.js'
 import { generateStores } from './stores.js'
 
-export default function viteFastifyReactPlugin () {
+export default function viteFastifyReactPlugin ({ ts } = {}) {
   const context = {
     root: null,
   }
   return [viteFastify({
-    clientModule: '$app/index.js'
+    clientModule: ts ? '$app/index.ts' : '$app/index.js'
```

Comment:
Does this mean the developer would have to explicitly pass `ts: true` to this plugin? I think we would want to avoid that and auto-detect this, right?

Reply:
I couldn't find a way to cleanly detect the intention of using the TypeScript virtual modules.

Right now it makes no difference anyway, the virtual modules are just the application shell, and their TypeScript version still has no types defined, they just have the `.ts` extension and the setup is ready to _run_ them (and also if they're _ejected_).